server/functions/__tests__/auth.spec.ts

```diff
@@ -0,0 +1,96 @@
+import mockProcess from './mocks/client/process';
+process = mockProcess;
+
+import { init } from './util/firebase';
+import { fail } from 'assert';
+import { expect, assert } from 'chai';
+import clientConfig from './mocks/client/config';
+import { GetTokenResult, IAuthController } from '../../../common/abstractions/controlllers/IAuthController';
+import { ClientAuthController } from './mocks/client/controllers';
+import { Users as UsersFunctions } from '../../../common/abstractions/functions';
+import Firebase, { initializeAsync } from '../../../common/services/firebase';
+import { validateToken } from '../src/auth';
+import * as userData from './mocks/data/users';
+
+// Initialize testing server
+const test = init('auth-test');
+
+// Tests
+let auth: IAuthController = null;
+describe('Authentication', () => {
+    beforeAll(async () => {
+        // Initialize testing client
+        await initializeAsync(clientConfig);
+    });
+    beforeEach(async () => {
+        await userData.create();
+        auth = new ClientAuthController();
+    });
+    afterEach(async () => {
+        auth = null;
+        await test.cleanup();
+        await userData.clear();
+    });
+    it('Should not generate an id token if client is not signed in', async () => {
+        const getTokenResult = await auth.getAuthToken();
+        assert.isFalse(getTokenResult.result);
+        assert.isUndefined(getTokenResult.token);
+    });
+    it('Should generate an id token for a logged in user', async () => {
+        const u = userData.getUser();
+        await auth.signInWithEmailPassword(u.email, u.password);
+        const getTokenResult: GetTokenResult = await auth.getAuthToken();
+        assert.isTrue(getTokenResult.result);
+        if (getTokenResult.token) {
+            console.log(`Token: ${getTokenResult.token}`);
+        } else {
+            fail();
+        }
+    });
+    it('Should not validate an invalid token', async () => {
+        const u = userData.getUser();
+        await auth.signInWithEmailPassword(u.email, u.password);
+        // const validate = await Firebase.Instance.getFunction(UsersFunctions.ValidateToken);
+        const args: any = {
+            type: 'validateToken',
+            token: 'notavalidtoken',
+            email: u.email,
+        };
+        // TODO: would like to invoke the function using the wrapper above
+        // currently experiencing issues in the emulation environment despite
```

Comment:
Tests pass now, but does this TODO (and the others in the file) need to be resolved?

Reply:
Let me know if the changes I made to the comments make sense. I'm going to avoid tackling these TODOs for the time being since they don't really affect the strength of the tests.